### PR TITLE
PP-10265 Update openapi spec parameter descriptions

### PR DESCRIPTION
**Pull request by james-peacock-gds**
- OpenAPI spec is being updated to be consistent with Pay API docs
- Working to [Nathan's changes](https://github.com/alphagov/pay-publicapi/pull/1617/files)
- Updated parameter descriptions to match the changes above

### CODE REVIEW

File: src/main/java/uk/gov/pay/api/resources/PaymentsResource.java [hunk 1]
```diff
@@ -195,32 +193,45 @@ public PaymentEventsResponse getPaymentEvents(@Parameter(hidden = true) @Auth Ac
...
+                                   @Parameter(description = "Returns payments in a matching `state`. `state` reflects where a payment is in the " +
+                                           "[payment status lifecycle](https://docs.payments.service.gov.uk/api_reference/#payment-status-lifecycle).", example = "success",
                                            schema = @Schema(allowableValues = {"created", "started", "submitted", "success", "failed", "cancelled", "error"}))
                                    @QueryParam("state") String state,
-                                   @Parameter(description = "Card brand used for payment. Example=master-card")
+                                   @Parameter(description = "Returns payments paid with a particular card brand.")
                                    @QueryParam("card_brand") String cardBrand,
-                                   @Parameter(description = "From date of payments to be searched (this date is inclusive). Example=2015-08-13T12:35:00Z")
+                                   @Parameter(description = "Returns payments created on or after the `from_date`. " +
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/api/resources/PaymentsResource.java:207]: In the format, `MM` is used to represent month as well as minute. Think it should be `YYYY-MM-DDTHH:mm:ssZ`

Can we still provide an example if there is a separate attribute in @Parameter?

File: src/main/java/uk/gov/pay/api/resources/SearchRefundsResource.java [hunk 2]
```diff
@@ -66,17 +66,27 @@ public SearchRefundsResource(SearchRefundsService searchRefundsService, PublicAp
     )
     public SearchRefundsResults searchRefunds(@Parameter(hidden = true)
                                   @Auth Account account,
-                                  @Parameter(description = "From date of refunds to be searched (this date is inclusive). Example=2015-08-13T12:35:00Z")
+                                  @Parameter(description = "Returns refunds created on or after the `from_date`. " +
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/api/resources/SearchRefundsResource.java:69]: Same as above for date format

File: src/main/java/uk/gov/pay/api/resources/SearchRefundsResource.java [hunk 3]
```diff
@@ -66,17 +66,27 @@ public SearchRefundsResource(SearchRefundsService searchRefundsService, PublicAp
...
+                                          "Date and time must use Coordinated Universal Time (UTC) and ISO 8601 format to second-level accuracy - `YYYY-MM-DDTHH:MM:SSZ`.")
                                   @QueryParam("from_date") String fromDate,
-                                  @Parameter(description = "To date of refunds to be searched (this date is exclusive). Example=2015-08-14T12:35:00Z")
+                                  @Parameter(description = "Returns refunds created before the `to_date`. " +
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/api/resources/SearchRefundsResource.java:72]: Same as above for date format

Comment (james-peacock-gds) [reply src/main/java/uk/gov/pay/api/resources/SearchRefundsResource.java:72]: thanks @kbottla I will feed that back to Nathan and get the changes made

File: src/main/java/uk/gov/pay/api/resources/PaymentsResource.java (hunk 1 above)
Comment (james-peacock-gds) [reply src/main/java/uk/gov/pay/api/resources/PaymentsResource.java:207]: date format updated. Nathan referred to https://www.w3.org/TR/NOTE-datetime and concluded the format should be YYYY-MM-DDThh:mm:ssZ

File: src/main/java/uk/gov/pay/api/resources/SearchRefundsResource.java (hunk 2 above)
Comment (james-peacock-gds) [reply src/main/java/uk/gov/pay/api/resources/SearchRefundsResource.java:69]: date format updated

File: src/main/java/uk/gov/pay/api/resources/SearchRefundsResource.java (hunk 3 above)
Comment (james-peacock-gds) [reply src/main/java/uk/gov/pay/api/resources/SearchRefundsResource.java:72]: date format updated

File: src/main/java/uk/gov/pay/api/resources/PaymentsResource.java [hunk 4]
```diff
@@ -195,32 +193,45 @@ public PaymentEventsResponse getPaymentEvents(@Parameter(hidden = true) @Auth Ac
...
-                                   @Parameter(description = "State of payments to be searched. Example=success", example = "success",
+                                   @Parameter(description = "Returns payments in a matching `state`. `state` reflects where a payment is in the " +
+                                           "[payment status lifecycle](https://docs.payments.service.gov.uk/api_reference/#payment-status-lifecycle).", example = "success",
                                            schema = @Schema(allowableValues = {"created", "started", "submitted", "success", "failed", "cancelled", "error"}))
                                    @QueryParam("state") String state,
-                                   @Parameter(description = "Card brand used for payment. Example=master-card")
+                                   @Parameter(description = "Returns payments paid with a particular card brand.")
                                    @QueryParam("card_brand") String cardBrand,
-                                   @Parameter(description = "From date of payments to be searched (this date is inclusive). Example=2015-08-13T12:35:00Z")
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/api/resources/PaymentsResource.java:207]: Can we add `example="2015-08-13T12:35:00Z" ` attribute as the example is no longer in the description

File: src/main/java/uk/gov/pay/api/resources/PaymentsResource.java [hunk 5]
```diff
@@ -195,32 +193,45 @@ public PaymentEventsResponse getPaymentEvents(@Parameter(hidden = true) @Auth Ac
...
+                                   @Parameter(description = "Returns payments created on or after the `from_date`. " +
+                                           "Date and time must be coordinated Universal Time (UTC) and ISO 8601 format to second-level accuracy - `YYYY-MM-DDThh:mm:ssZ`.")
                                    @QueryParam("from_date") String fromDate,
-                                   @Parameter(description = "To date of payments to be searched (this date is exclusive). Example=2015-08-14T12:35:00Z")
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/api/resources/PaymentsResource.java:209]: Same here for example

Comment (james-peacock-gds) [reply src/main/java/uk/gov/pay/api/resources/PaymentsResource.java:209]: done

File: src/main/java/uk/gov/pay/api/resources/SearchRefundsResource.java [hunk 6]
```diff
@@ -66,17 +66,27 @@ public SearchRefundsResource(SearchRefundsService searchRefundsService, PublicAp
...
+                                  @Parameter(description = "Returns refunds created on or after the `from_date`. " +
+                                          "Date and time must use Coordinated Universal Time (UTC) and ISO 8601 format to second-level accuracy - `YYYY-MM-DDThh:mm:ssZ`.", example = "2015-08-13T12:35:00Z")
                                   @QueryParam("from_date") String fromDate,
-                                  @Parameter(description = "To date of refunds to be searched (this date is exclusive). Example=2015-08-14T12:35:00Z")
+                                  @Parameter(description = "Returns refunds created before the `to_date`. " +
+                                          "Date and time must use Coordinated Universal Time (UTC) and ISO 8601 format to second-level accuracy - `YYYY-MM-DDThh:mm:ssZ`.", example = "2015-08-13T12:35:00Z")
                                   @QueryParam("to_date") String toDate,
-                                  @Parameter(description = "From settled date of refund to be searched (this date is inclusive). Example=2015-08-13")
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/api/resources/SearchRefundsResource.java:73]: Can we add the `example` attribute here too?

File: src/main/java/uk/gov/pay/api/resources/SearchRefundsResource.java [hunk 7]
```diff
@@ -66,17 +66,27 @@ public SearchRefundsResource(SearchRefundsService searchRefundsService, PublicAp
...
+                                  @Parameter(description = "Returns refunds settled on or after the `from_settled_date` value. " +
+                                          "You can only use `from_settled_date` if your payment service provider is Stripe. " +
+                                          "Date must use ISO 8601 format to date-level accuracy - `YYYY-MM-DD`. " +
+                                          "Refunds are settled when Stripe takes the refund from your account balance.")
                                   @QueryParam("from_settled_date") String fromSettledDate,
-                                  @Parameter(description = "To settled date of refund to be searched (this date is inclusive). Example=2015-08-13")
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/api/resources/SearchRefundsResource.java:75]: Same here for example

File: src/main/java/uk/gov/pay/api/resources/SearchRefundsResource.java (hunk 6 above)
Comment (james-peacock-gds) [reply src/main/java/uk/gov/pay/api/resources/SearchRefundsResource.java:73]: done

File: src/main/java/uk/gov/pay/api/resources/SearchRefundsResource.java (hunk 7 above)
Comment (james-peacock-gds) [reply src/main/java/uk/gov/pay/api/resources/SearchRefundsResource.java:75]: done

**Review by kbottla — APPROVED** [review]

**Pull request merged**